<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script src="./promise.js"></script>
  <script>
    /* var promise = new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve('success');
        }, 100);
    })
    promise.then((val) => {
      console.log(val);
      return 222
    }).then(console.log) */

    // console.log(Promise.resolve(2).finally(() => {}));

    // promise吃掉错误，因为执行excute的时候会try..catch
    let promise = new Promise(() => {
        throw new Error('error')
    });
    console.log(2333333);
  </script>
</body>
</html>